test/e2e/karmadactl_test.go

```diff
@@ -610,3 +610,148 @@ var _ = ginkgo.Describe("Karmadactl top testing", ginkgo.Labels{NeedCreateCluste
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Karmadactl logs testing", func() {
+	var member1 string
+	var member1Client kubernetes.Interface
+
+	ginkgo.BeforeEach(func() {
+		member1 = "member1"
+		member1Client = framework.GetClusterClient(member1)
+	})
+
+	waitForPodReady := func(namespace, podName string) {
+		gomega.Eventually(func() bool {
+			pod, err := member1Client.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+			if err != nil {
+				return false
+			}
+			return pod.Status.Phase == corev1.PodRunning
+		}, pollTimeout, pollInterval).Should(gomega.BeTrue(), fmt.Sprintf("Pod %s in namespace %s is not running", podName, namespace))
+	}
```

Comment:
```suggestion
	waitForPodReady := func(namespace, podName string) {
			framework.WaitPodPresentOnClusterFitWith(framework.ClusterNames()[0], namespace, podName, func(pod *corev1.Pod) bool {
				return pod.Status.Phase == corev1.PodRunning
			})
		}
```

Reply:
That is definitely much better! What about using the local member cluster instead in the test case to make it independent of the ordering of cluster names i.e. `member1` instead of `framework.ClusterNames()[0]`?

Reply:
Or another suggestion is replacing all `member<i>` hard coded in the all test cases with `framework.ClusterNames()[i]`?

Reply:
> Or another suggestion is replacing all `member<i>` hard coded in the all test cases (not only log testing) with `framework.ClusterNames()[i]`?

I think it’s no need to expand the scope of modifications.

Reply:
Makes sense, Thanks!

Reply:
> Or another suggestion is replacing all member<i> hard coded in the all test cases (not only log testing) with framework.ClusterNames()[i]?

Hi @mohamedawnallah, that is what I want, can you help do it in a separate PR?

Reply:
Sure, I'd be happy to help with that. I'll create a separate PR for this :)

Reply:
> > Or another suggestion is replacing all member hard coded in the all test cases (not only log testing) with framework.ClusterNames()[i]?
> 
> Hi @mohamedawnallah, that is what I want, can you help do it in a separate PR?

@XiShanYongYe-Chang, I submitted a PR for this issue #5226 :)